megalodon/src/megalodon.ts

```diff
@@ -1352,16 +1355,26 @@ export const detector = async (url: string, proxyConfig: ProxyConfig | false = f
     })
   }
 
-  const res = await axios.get<Instance>(url + '/api/v1/instance', options)
-  if (res.data.title) {
-    if (res.data.pleroma) {
-      return 'pleroma'
+  try {
+    const res = await axios.get<Instance>(url + '/api/v1/instance', options)
+    if (res.data.title) {
+      if (res.data.pleroma) {
+        return 'pleroma'
+      } else {
+        return 'mastodon'
+      }
     } else {
-      return 'mastodon'
+      return null
```

Comment:
I think when the URL is `unknown`, it should reject (or Exception?) because this function is `async`. Why do you return `null`?

Reply:
Yes I believe you made total sense. Code has been updated to throw instead of returning a `null`./